python/dalex/dalex/fairness/_group_fairness/utils.py

```diff
@@ -213,29 +217,172 @@ def _metric_ratios_2_df(fobject):
+            }
+
+        for subgroup in self.regression_dict.keys():
+
+            self.subgroup_metric_comparison[subgroup] = {
+                'mae_ratio': self.subgroup_metrics[subgroup].get("mae") / self.subgroup_metrics[privileged].get("mae"),
+                'rmse_ratio': self.subgroup_metrics[subgroup].get("rmse") / self.subgroup_metrics[privileged].get("rmse"),
+                'mean_prediction_ratio': self.subgroup_metrics[subgroup].get("mean_prediction") / self.subgroup_metrics[
+                    privileged].get("mean_prediction"),
+            }
+
+    def __str__(self):
+        return f"Fairness in Regression Dictionary\nSubgroup Metrics: {self.subgroup_metrics.values()}\n" \
+               f"Subgroup Metric Comparison: {self.subgroup_metric_comparison.values()}"
+
+# -------------- Functions needed in creation and methods of GroupFairnessRegression object in object.py --------------
+
+def calculate_regression_measures(y, y_hat, protected, privileged):
+
+    unique_protected = np.unique(protected)
+    unique_unprivileged = unique_protected[unique_protected != privileged]
+
+    data = pd.DataFrame(columns=['subgroup', 'independence', 'separation', 'sufficiency'])
+
+    for unprivileged in unique_unprivileged:
+        # filter elements
+        array_elements = np.isin(protected, [privileged, unprivileged])
+
+        y_scaler = StandardScaler()
+        y_hat_scaler = StandardScaler()
+
+        y_u = y_scaler.fit_transform(y[array_elements].reshape(-1, 1))
+        s_u = y_hat_scaler.fit_transform(y_hat[array_elements].reshape(-1, 1))
+
+        a = np.where(protected[array_elements] == privileged, 1, 0)
+
+        p_s = LogisticRegression()
+        p_ys = LogisticRegression()
+        p_y = LogisticRegression()
+
+        p_s.fit(s_u, a)
+        p_y.fit(y_u, a)
+        p_ys.fit(np.c_[y_u, s_u], a)
+
+        pred_p_s = p_s.predict_proba(s_u.reshape(-1, 1))[:, 1]
+        pred_p_y = p_y.predict_proba(y_u.reshape(-1, 1))[:, 1]
+        pred_p_ys = p_ys.predict_proba(np.c_[y_u, s_u])[:, 1]
+
+        n = len(a)
+
+        r_ind = ((n - a.sum()) / a.sum()) * (pred_p_s / (1 - pred_p_s)).mean()
+        r_sep = ((pred_p_ys / (1 - pred_p_ys) * (1 - pred_p_y) / pred_p_y)).mean()
+        r_suf = ((pred_p_ys / (1 - pred_p_ys)) * ((1 - pred_p_s) / pred_p_s)).mean()
+
+        to_append = pd.DataFrame({'subgroup': [unprivileged],
+                                  'independence': [r_ind],
+                                  'separation': [r_sep],
+                                  'sufficiency': [r_suf]})
+
+        data = data.append(to_append)
+
+    # append the scale
+    to_append = pd.DataFrame({'subgroup': [privileged],
+                              'independence': [1],
+                              'separation': [1],
+                              'sufficiency': [1]})
+
+    data.index = data.subgroup
+    data = data.iloc[:,1:]
+    return data
+
+
 # -------------- Helper functions --------------
```

Comment:
If the functions below are checks, don't they belong in the `checks.py` file?

e.g. `checks.check_epsilon` is used in the `universal_fairness_check` - these could be next to each other.

Reply:
These functions have check in the name but they are helper functions. For example `universal_fairness_check` is a function that acts as a fairness check for both regression and classification.